src/util/file-filter.js

```diff
@@ -0,0 +1,87 @@
+/* @flow */
```

Comment:
I found some bugs when writing tests for FileFilter.
```
    it('ignore artifactsDir and its content', () => {
      const filter = new FileFilter({
        artifactsDir: 'artifacts',
      });
      assert.equal(filter.wantFile('artifacts'), false);
      assert.equal(filter.wantFile('artifacts/some.js'), false); // assertion error
    });
```
It might be fine for `build.js`, since the `zipDir` will skip the content if the parent dir is skipped.

But it cause problem for `watcher.js`. Make the content inside `artifactsDir` becomes wanted files. This bug applies to `.git` and `node_modules` folders as well.